core/dbt/contracts/graph/nodes.py

```diff
@@ -492,6 +492,60 @@ def search_name(self):
     def materialization_enforces_constraints(self) -> bool:
         return self.config.materialized in ["table", "incremental"]
 
+    def infer_primary_key(self, data_tests) -> List[str]:
```

Comment:
is it possible to type annotate this?

Reply:
I get this error when I try to type it:
<img width="1021" alt="Screenshot 2024-02-23 at 12 11 18 PM" src="https://github.com/dbt-labs/dbt-core/assets/2550003/9d07b4d8-212a-40b6-b655-780c8ae43531">

I think this is because GenericTestNode is defined in the same file but after ModelNode. We could define the GenericTestNode in a separate file and import it instead, but that's probably better to do in a separate PR and for all the node types?

Reply:
I think quoting it should do the trick!

e.g. `data_tests: List["GenericTestNode"]`

Reply:
Thanks! Updated